### PR TITLE
Feature/real time ais

### PR DESCRIPTION
**Pull request by RicardoAndreSantos**
Ais real time updates.

### CODE REVIEW

File: src/main/java/pt/lsts/ripples/controllers/WebSocketsController.java [hunk 1]
```diff
@@ -24,7 +24,6 @@ public WebSocketsController(SimpMessagingTemplate template) {
 
 	// the business logic can call this to update all connected clients
     public void sendAssetUpdateFromServerToClients(Asset asset) {
-        // logger.info("Broadcasting update for asset " + asset.getName());
```
Comment (zepinto) [inline src/main/java/pt/lsts/ripples/controllers/WebSocketsController.java:27]: Please remove commented code.

File: src/main/java/pt/lsts/ripples/controllers/NeptusController.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+
+@Controller
+public class NeptusController {
```
Comment (zepinto) [inline src/main/java/pt/lsts/ripples/controllers/NeptusController.java:11]: Is this controller being used? if not it should not be part of the pull request.

File: src/main/java/pt/lsts/ripples/services/AisFetcherService.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    ByteArrayOutputStream out = new ByteArrayOutputStream();;
+
+
+    @Autowired
+    AISRepository repo;
+    
+    @Autowired
+	WebSocketsController wsController;
+
+    @Value("${udp.port}")
```
Comment (zepinto) [inline src/main/java/pt/lsts/ripples/services/AisFetcherService.java:48]: Please provide a default value (https://www.baeldung.com/spring-value-defaults).

File: src/main/java/pt/lsts/ripples/services/AisFetcherService.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    private int port; 
+
+    @PostConstruct
+    public void init() throws IOException {
+        logger.info("Started UPD listener on port: " + port);
```
Comment (zepinto) [inline src/main/java/pt/lsts/ripples/services/AisFetcherService.java:53]: Typo in UDP.

File: src/main/java/pt/lsts/ripples/services/AisFetcherService.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+        UdpListenner client = new UdpListenner();
+        ExecutorService exec = Executors.newFixedThreadPool(1);
+        exec.submit(client);
+    }
+
+    public class UdpListenner implements Runnable {
```
Comment (zepinto) [inline src/main/java/pt/lsts/ripples/services/AisFetcherService.java:60]: Class name has a typo (UdpListenner -> UDPListener)

**Review by zepinto — CHANGES_REQUESTED** [review]
Please check the comments inline.

File: src/main/java/pt/lsts/ripples/controllers/NeptusController.java (hunk 2 above)
Comment (RicardoAndreSantos) [reply src/main/java/pt/lsts/ripples/controllers/NeptusController.java:11]: Not yet, it was just an experience for the communication with Neptus.

File: src/main/java/pt/lsts/ripples/controllers/WebSocketsController.java [hunk 6]
```diff
@@ -22,7 +22,6 @@ public WebSocketsController(SimpMessagingTemplate template) {
         this.template = template;
     }
 
-	// the business logic can call this to update all connected clients
```
Comment (zepinto) [inline src/main/java/pt/lsts/ripples/controllers/WebSocketsController.java:25]: Documentation is welcome. Just asked to remove the commented out code.

**Review by zepinto — APPROVED** [review]

**Pull request merged**